src/main/java/bowling/domain/Frames.java

```diff
@@ -0,0 +1,206 @@
+package bowling.domain;
+
+import bowling.Controller;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class Frames {
+    public static final int MAP_ROUND_TO_INDEX_CONSTANT = 1;
+
+    private final List<Frame> frames = new ArrayList<>();
+    private final String player;
+
+    public Frames(String player) {
+        this.player = player;
+    }
+
+    public boolean playFrame(final int numberOfFallenPins, final Frame newFrame) {
+        addFrame(newFrame);
+        Frame frame = play(numberOfFallenPins);
+        return frame.endFrame();
+    }
+
+    private void addFrame(final Frame frame) {
+        if (frames.size() == 0) {
+            frames.add(frame);
+        }
+        if (lastFrame().endFrame()) {
+            frames.add(frame);
+        }
+    }
+
+    private Frame play(final int numberOfFallenPins) {
+        return lastFrame().play(lastIndex(), numberOfFallenPins);
+    }
```

Comment:
```suggestion
    public void addFrame(final Frame frame) {
        if (frames.size() == 0) {
            frames.add(frame);
        }
        if (lastFrame().endFrame()) {
            frames.add(frame);
        }
    }

    private Frame play(final int numberOfFallenPins) {
        return lastFrame().play(lastIndex(), numberOfFallenPins);
    }
```

하지만 play 를 통해 다음 Frame 을 생성하는 로직은,
이전에 Bowl 한 State 에 따라서 달라지게 된다고 생각합니다

저는 play 부분을 외부의 다른 객체에게 위임하고,
Frames 는 컬렉션의 기본적인 역할에 맞게, addFrame 메서드를 외부에 public 하게 공개하는 것으로 충분하지 않나? 라고 생각이 듭니다.